code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -28,47 +28,43 @@
 	return ..()
 
 /mob/living/simple_animal/hostile/proc/FindTarget()
-
 	if(!faction) //No faction, no reason to attack anybody.
 		return null
 
 	var/atom/T = null
 	var/lowest_health = INFINITY // Max you can get
 	stop_automated_movement = 0
-
-	for(var/atom/A in targets)
-
-		if(A == src)
+	for(var/mob/living/L in targets)

```

Comment:
I'd like to see this loop removed. You're iterating the `targets` list twice. You can just prioritize livings in the main loop and break immediately as you find one.